transport.go

```diff
@@ -204,7 +205,28 @@ func (t *Transport) DialEarly(ctx context.Context, addr net.Addr, tlsConf *tls.C
 	return t.dial(ctx, addr, "", tlsConf, conf, true)
 }
 
-func (t *Transport) dial(ctx context.Context, addr net.Addr, host string, tlsConf *tls.Config, conf *Config, use0RTT bool) (EarlyConnection, error) {
+func (t *Transport) dial(ctx context.Context, addr net.Addr, host string, tlsConf *tls.Config, conf *Config, use0RTT bool) (c EarlyConnection, err error) {
+	trace := httptrace.ContextClientTrace(ctx)
+	if trace != nil {
+		if trace.ConnectStart != nil {
+			trace.ConnectStart(addr.Network(), addr.String())
+		}
+		if trace.TLSHandshakeStart != nil {
+			trace.TLSHandshakeStart()
+		}
```

Comment:
For consistency, this should be a `traceConnectStart` and `traceTLSHandshakeStart` in trace.go, so you can avoid the `if` statements here.